sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/CatalystQlSuite.scala

```diff
@@ -171,6 +171,7 @@ class CatalystQlSuite extends PlanTest {
   test("table identifier") {
     assert(TableIdentifier("q") === parser.parseTableIdentifier("q"))
     assert(TableIdentifier("q", Some("d")) === parser.parseTableIdentifier("d.q"))
+    assert(TableIdentifier("104e4d676bac4d9aa3856f00b5b9f51c") === parser.parseTableIdentifier("104e4d676bac4d9aa3856f00b5b9f51c"))
```

Comment:
You are not testing the AbstractSQLParser here but CatalystQl (which probably supports such a table identifier).